nexus_motion_planner/src/motion_planner_server.hpp

```diff
@@ -88,6 +95,19 @@ class MotionPlannerServer : public rclcpp_lifecycle::LifecycleNode
   double _workspace_max_y;
   double _workspace_max_z;
 
+  // Motion plan caching
+  std::shared_ptr<nexus::motion_planner::MotionPlanCache> _motion_plan_cache;
+
+  bool _use_motion_plan_cache;
```

Comment:
Instead of three bools, let's define class which defines how the databse will be used for better extensibility.
```
enum class PlannerDatabaseMode: public unint8_t
{
  // The planner will generate a plan. Find if an entry exists in the databse and overwrite if execution time is better.
  Training,
  
  // Try to lookup a cached plan and if non is available, plan live but do not update the cache.
  Optional,

  // Return a plan only if present in the database cache.
  // ReadOnly, 
};

```

And below we would store an optional of this type.
```
std::optional<PlannerDatabaseMode> = std::nullopt;
```

We can parse a similarly named ROS param which takes in a string with the same name as the enum variants above. If we have a matching name, we set the member to the appropriate enum; Else leave it as optional. In the service response, if this member is nullopt, do a live plan (same as now). Else if not nullopt, plan and update database accordingly.
We should also instantiate the `_motion_plan_cache` only if this variable is not nullopt and other database specific parameters are defined (port, etc).

Reply:
d85907d118b82ffd8ab1efb798f8b71d9c66fc78